plugins/mcreator-core/blockly/js/field_data_list_dropdown.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * This class represents a data list field that can be used to choose value from potentially short data lists.
+ */
+class FieldDataListDropdown extends Blockly.FieldDropdown {
+
+    EDITABLE = true;
+    SERIALIZABLE = true;
+    CURSOR = 'default';
+
+    constructor(datalist = '', opt_validator, opt_config) {
+        super(function () {
+            return arrayToBlocklyDropDownArray(javabridge.getListOf(datalist));
+        }, opt_validator, opt_config);
+        this.type = datalist;
+        this.maxDisplayLength = 75;
+    };
+
+    // Configure the field given a map of settings
+    // Create the field from the json definition
+    static fromJson(options) {
+        return new this(Blockly.utils.parsing.replaceMessageReferences(options['datalist']), undefined, options);
+    };
+
+    // Update the value of this selector
+    doValueUpdate_(newValue) {
+        if (newValue !== this.value_) { // If the value is different, update the readable name
+            if (newValue) { // Try to get the readable name from the value
+                this.readableName = javabridge.getReadableNameOf(newValue, this.type) || newValue;
+            } else { // Fallback to the "No entry selected" message
+                this.readableName = FieldDataListSelector.getDefaultText();

```

Comment:
`FieldDataListSelector.getDefaultText()` doesn't exist. `newValue` also always exists, since you're pulling values from a (supposedly non empty) datalist